lib/salestation/web/extractors.rb

```diff
@@ -192,35 +193,21 @@ def self.[](filters:, rack_key:)
         end
 
         def self.extract(filters, request_hash)
-          filters.each_with_object({}) do |filter, extracted_data|
-            case filter
-            when Symbol
-              stringified_key = filter.to_s
-              extracted_data[filter] = request_hash[stringified_key] if request_hash.key?(stringified_key)
-              if extracted_data[filter].is_a?(Array)
-                extracted_data[filter] = extracted_data[filter].map{ |hash|
-                  if hash.is_a?(Hash)
-                    hash.transform_keys(&:to_sym) if hash.is_a?(Hash)
-                  else
-                    hash
-                  end
-                }
-              end
-            when Hash
-              filter.each do |key, nested_filters|
-                stringified_key = key.to_s
-                if request_hash.key?(stringified_key)
-                  value = request_hash.fetch(stringified_key)
-                  extracted_data[key] =
-                    if !value.is_a?(Hash)
-                      value
-                    else
-                      extract(nested_filters, value)
-                    end
-                end
-              end
-            end
+          # Filter as a hash is used in some existing implementations that did not expect full 
+          # recursive symbolizing of keys. In this case hash objects at the highest level of object 
+          # are represented as hash of filter keys. This is no longer needed, but we support it 
+          # to avoid a breaking change.
+
+          filters_flat = filters
+                          .map{|filter| filter.is_a?(Hash) ? filter.keys : filter}
+                          .flatten()
+                          .map{|filter| filter.to_s}
+
+          request_hash = request_hash.select do
```

Comment:
Looks like we don't have rubocop here. `|k, v|` should be on this line:

```
request_hash = request_hash.select do |k,v|
  filters_flat.include?(k)
end
```

The one liner should look like this:
```
request_hash = request_hash.select {|k,v| filters_flat.include?(k)}
```

Reply:
Very nice! I made the change. And yes, I ran rubocop locally in the project - it shows 479 issues 😄 So most probably it is not included in the pipeline. I will at the moment ignore this, as this is not related to my changes.